blocks/api/parser.js

```diff
@@ -177,6 +178,12 @@ export function createBlockWithFallback( name, rawContent, attributes ) {
 	let blockType = getBlockType( name );
 	const fallbackBlock = getUnknownTypeHandlerName();
 	if ( ! blockType ) {
+		// If detected as a block which is not registered, preserve comment
+		// delimiters in content of unknown type handler.
+		if ( name ) {
+			rawContent = getCommentDelimitedContent( name, attributes, rawContent );
```

Comment:
Why do we need to manually preserve it?

Reply:
The result of a parse doesn't include the comment demarcations. Take this example:

```
<!-- wp:core/unknown -->
Unknown block
<!-- /wp:core/unknown -->
```

Upon parse, this is passed to `createBlockWithFallback` with arguments: `core/unknown` (`name`), `Unknown block` (`rawContent`), `{ foo: 'bar' }` (`attributes`). When it fails to find the block matching this slug, the `name` and behavior assumes that of the fallback block. Future reserialization will not know to save this as the `core/unknown` block. So the logic here re-adds the comments to the `rawContent` field, so even though this assumes the role of the fallback block, the fallback block's content includes the original block text, and since the fallback block no longer writes its own comment demarcation, it is saved exactly as it was originally.

Part of this assumes we want the freeform block behaviors to be made available for unknown blocks. The original pull request comment explores whether this is desirable, or if we'd want to just flag it as "unknown", leave it alone, and notify the user as such.

>Because Freeform renders a TinyMCE, it is very easy to mistakenly edit the content of an unknown block type. I'm wondering if we ought to either create a separate block representation for un-blocked content vs. block content of an unknown type (e.g. since-disabled plugins blocks).
>
>Alternatively, or in either case, we may want to convey messaging to the user about the unknown block instead of providing them the option to edit.

Reply:
Thanks for the explanation. The previous behaviour would transform the unknown block to freeform but still retain the comment within it, hence my question.

Reply:
>The previous behaviour would transform the unknown block to freeform but still retain the comment within it

No, the problem was that it didn't retain the comment. The logic added here is what achieves this.